debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/gccheck/CheckEngine.java

```diff
@@ -1189,6 +1194,29 @@ private int checkJ9Object(J9ObjectPointer object, GCHeapRegionDescriptor regionD
 				return ret;
 			}
 		}
+
+		if (GCExtensions.isVLHGC() && J9BuildFlags.env_data64 && isIndexableDataAddressFlagSet() && ObjectModel.isIndexable(object)) {
+			try {
+				boolean isCorrectData = false;
+				J9IndexableObjectPointer array = J9IndexableObjectPointer.cast(object);
+				if (J9IndexableObjectHelper.isInlineContiguousArraylet(array)) {
+					isCorrectData = J9IndexableObjectHelper.getDataAddrForContiguous(array).equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.contiguousHeaderSize())));
+				} else {
+					isCorrectData = J9IndexableObjectHelper.getDataAddrForDiscontiguous(array).equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.discontiguousHeaderSize())));
+				}
+
+				if (false == isCorrectData) {
+					return J9MODRON_GCCHK_RC_INVALID_INDEXABLE_DATA_ADDRESS;
+				}
+			} catch (NoSuchFieldException e) {
```

Comment:
The body of the `try` block should be reduced the only the part that might throw `NoSuchFieldException`.